static/src/stylesheets/head.ad-free.scss

```diff
@@ -0,0 +1,9 @@
+// this is slightly kludgy way of clearing up the space left by ad slots in
+// ad-free. it used to be done in the commercial JS, but that's not loaded in
+// ad-free now. ideally we'd not render the slots in the first place, but the
+// ad-free path is slightly labyrinthine, and soon DCR will handle fronts
+// anyway.
+
+.ad-slot, .top-banner-ad-container {
+    display: none !important;
```

Comment:
Agreed this is quite a blunt way of removing slots for adfree but we can live with it until the fronts migration.

We can also take an action to document what conditions contribute to `adFree` in case there is a gap on DCR.

Reply:
one example of this:

to force ad-free I prepended this line with `true || `: https://github.com/guardian/frontend/blob/2d276806f63ae0301ccfb956bf4a360d5218739d/common/app/views/support/Commercial.scala#L17

that left me with a container that could have an MPU that looked like this: 
<img width="954" alt="image" src="https://user-images.githubusercontent.com/867233/222454165-e5694834-6210-445f-a80a-9bbb20cbe829.png">

but for a real logged in, ad-free user, you get this: 
<img width="959" alt="image" src="https://user-images.githubusercontent.com/867233/222454393-a8544b63-53e7-4e5f-95e4-05dd34056404.png">

i really struggled to work out just how that slice was being rendered differently. i thought it might be as simple as checking for ad-free wherever we check for the commercial switch too, but afaict, not all ad slots are actually wrapped in that check. the same is true for `shouldShowAds` afaict 

https://github.com/guardian/frontend/blob/2d276806f63ae0301ccfb956bf4a360d5218739d/common/app/views/support/Commercial.scala#L29

I kind of expected all 3 of these to always be checked in the same places, but they weren't. I'm sure they could be rationalised along the lines of DCR: https://github.com/guardian/dotcom-rendering/blob/933a21ec5faabb0b723ac203adca5a1b7386bddc/dotcom-rendering/src/web/layouts/StandardLayout.tsx#L343

but yeah, it seemed a lot to do (and get correct) when it would only apply for a short while

Reply:
Thanks for that example. I think we have a mix of inputs:

- server side: switches, sensitive content
- client side: subscription cookies, request overrides
- client side piggybacking: e.g. setting adfree for when the user rejects consent

I'll raise a ticket for commercial to review the requirements and where we can reduce fidelity in exchange for simplicity.

Or at least centralise to a lib which would make the inputs explicit.